src/plugins/intel_gpu/src/plugin/ops/eye.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph/op/eye.hpp"
+
+#include <memory>
+
+#include "intel_gpu/plugin/common_utils.hpp"
+#include "intel_gpu/plugin/program.hpp"
+#include "intel_gpu/primitives/eye.hpp"
+
+namespace ov {
+namespace runtime {
+namespace intel_gpu {
+
+namespace {
+
+static void CreateEyeOp(Program& p, const std::shared_ptr<ngraph::op::v9::Eye>& op) {
+    p.ValidateInputs(op, {3, 4});
+
+    const InferenceEngine::SizeVector& output_shapes = op->get_output_shape(0);
+    auto os_sz = output_shapes.size();
+    assert(2 <= os_sz && os_sz <= 5);
```

Comment:
Maybe better to use throw or IE_THROW here instead of assert?

Reply:
that is debugging guard, the base class (ngraph::op::v9::Eye) should throw an exception if there is the wrong output size.